pyanaconda/installation.py

```diff
@@ -95,7 +97,26 @@ def doConfiguration(storage, payload, ksdata):
     os_config = TaskQueue("Installed system configuration", N_("Configuring installed system"))
     os_config.append(Task("Configure authselect", ksdata.authselect.execute))
     os_config.append(Task("Configure SELinux", ksdata.selinux.execute))
-    os_config.append(Task("Configure first boot tasks", ksdata.firstboot.execute))
+
+    services_proxy = SERVICES.get_proxy()
+    services_dbus_tasks = services_proxy.InstallWithTasks(util.getSysroot())
+    # add one Task instance per DBUS task
+    for dbus_task in services_dbus_tasks:
+        task_proxy = SERVICES.get_proxy(dbus_task)
+        os_config.append(Task(task_proxy.Name, sync_run_task, (task_proxy,)))
+
+    # If Initial Setup is disabled notify Screen Access Manager so that
+    # this information ends up correctly in user interaction config file.
+    # Otherwise "firstboot --disable" in kickstart would only disable Initial Setup
+    # but not other tools reading the config file, such as Gnome Initial Setup.
+    if services_proxy.SetupOnBoot == SETUP_ON_BOOT_DISABLED:
+        # we need a simple function that does the assignment we can put into a task
+        def mark_post_inst_tools_disabled():
+            screen_access.sam.post_install_tools_disabled = True
```

Comment:
I would like to see this code in `screen_access.py`. For example, in a new method `update_config_file`.

Reply:
So basically a hook that would be called from installation.py ? The hook would then check the SetupOnBootState and set post_install_tools disabled accordingly. Does that sound right ?

Reply:
Yes :-)